src/Service/CoverStore/CoverStoreTransformationInterface.php

```diff
@@ -20,25 +20,49 @@ interface CoverStoreTransformationInterface
      *
      * @param string $url
      *   The base URL to transform
+     * @param int $width
+     *   The original image width for the url linked to
+     * @param int $height
+     *   The original image height for the url linked to
      * @param string $format
      *   The format to use
      *
-     * @return string
+     * @return string|null
+     *   If the transformation has bigger dimensions than the original image null will be returned
      *
      * @throws CoverStoreTransformationException
      */
-    public function transform(string $url, string $format = 'default'): string;
+    public function transform(string $url, int $width, int $height, string $format = 'default'): ?string;
 
     /**
      * Apply all configured transformations.
      *
      * @param string $url
      *   The base URL to transform
+     * @param int $width
+     *   The original image width for the url linked to
+     * @param int $height
+     *   The original image height for the url linked to
      *
      * @return array
-     *   All transformation keyed by name
+     *   All transformation keyed by name (If the transformation has bigger dimensions than the original image null will
```

Comment:
All transformation**s**

Reply:
Thats still correct now some values in the array maybe null and not an URL

Reply:
Just add an 's' to the end of transformation :)